### PR TITLE
Add arm64 support for env2yaml

### PR DESCRIPTION
**Pull request by robbavey**
This commit builds env2yaml in arm64 and amd64 flavors, and uses $TARGETARCH in the Dockerfile to ensure that the correct version is used when building for alternative architectures

Fixes: #15913

### CODE REVIEW

**Comment by robbavey** [comment]
Buildkite docker test runs - 

amd64 - https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/279#018e19c4-eaa2-4ed7-bba3-34c541bb9c6c

arm64 - https://buildkite.com/elastic/logstash-aarch64-pipeline/builds/62#018e19fb-8ebe-4985-ab11-fe71ed7c0297

**Comment by robbavey** [comment]
Buildkite - amd64 acceptance - https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/280#018e1a4d-869a-4c11-a477-49ef604e76cc
Buildkite - arm64 acceptance - https://buildkite.com/elastic/logstash-aarch64-pipeline/builds/63#018e1a4d-a364-4ee0-a6e7-de7f3143f061

Testing Notes - This should create docker images that work for `arm64` and `aarch64` architectures, and running `rake artifact:docker` should create a working docker image when run on the appropriate architecture.

This commit also replaces the `env2yaml` binary with `env2yaml-arm64` and `env2yaml-amd64` binaries in the docker build context, and uses `$TARGETARCH` to determine which architecture to use, which should reinstate compatibility on docker images supplied on `dockerhub`

File: docker/Makefile [hunk 1]
```diff
@@ -56,7 +56,8 @@ build-from-local-ubi8-artifacts: dockerfile env2yaml
 
 COPY_FILES := $(ARTIFACTS_DIR)/docker/config/pipelines.yml $(ARTIFACTS_DIR)/docker/config/logstash-oss.yml $(ARTIFACTS_DIR)/docker/config/logstash-full.yml
 COPY_FILES += $(ARTIFACTS_DIR)/docker/config/log4j2.file.properties $(ARTIFACTS_DIR)/docker/config/log4j2.properties
-COPY_FILES += $(ARTIFACTS_DIR)/docker/pipeline/default.conf $(ARTIFACTS_DIR)/docker/bin/docker-entrypoint $(ARTIFACTS_DIR)/docker/env2yaml/env2yaml
+COPY_FILES += $(ARTIFACTS_DIR)/docker/pipeline/default.conf $(ARTIFACTS_DIR)/docker/bin/docker-entrypoint
+COPY_FILES += $(ARTIFACTS_DIR)/docker/env2yaml/env2yaml-arm64 $(ARTIFACTS_DIR)/docker/env2yaml/env2yaml-amd64
```
Comment (andsel) [inline docker/Makefile:60]:
```suggestion
COPY_FILES += $(ARTIFACTS_DIR)/docker/env2yaml/env2yaml-arm64 
COPY_FILES += $(ARTIFACTS_DIR)/docker/env2yaml/env2yaml-amd64
```

If possibile I would split in 2 lines to make it more evident there are 2 architectures

**Review by andsel — APPROVED** [review]
LGTM, left a really minor comment to improve readability

**Comment by robbavey** [comment]
@logstashmachine backport 8.13

**Pull request merged**